OpenRA.Mods.Common/Traits/Buildings/Reservable.cs

```diff
@@ -29,11 +29,18 @@ public void Tick(Actor self)
 				return;		/* nothing to do */
 
 			if (!Target.FromActor(reservedFor).IsValidFor(self))
-				reservedFor = null;		/* not likely to arrive now. */
+			{
+				/* not likely to arrive now. */
+				reservedFor = null;
+				reservedForAircraft = null;
```

Comment:
I wonder if not calling `reservedForAircraft.UnReserve();` here might cause your reservation issue.